integration_tests/juniper_tests/src/codegen/impl_scalar.rs

```diff
@@ -0,0 +1,42 @@
+use fnv::FnvHashMap;
+#[cfg(test)]
+use juniper::{
+    self, parser::ScalarToken, DefaultScalarValue, GraphQLType, InputValue, ParseScalarResult,
+    ParseScalarValue, Value,
+};
+
+struct UserId(String);
+
+// TODO Trait that the macro handles, move to proper location! Naming?
+trait ParseCustomScalarValue<T, S = DefaultScalarValue> {
+    fn resolve(&self) -> Value;
+    fn from_input_value(value: &InputValue) -> Option<T>;
+    fn from_str<'a>(value: ScalarToken<'a>) -> ParseScalarResult<'a, S>;
+}
+
+#[juniper::graphql_scalar2(name = "MyCustomName", description = "My custom description")]
```

Comment:
You can just delete the `graphql_scalar!`, rename this, and macro and update all usages in tests. It might find some regressions as well.